Add external configuration section

griffinhadfield opened the pull request:
Addressing https://github.com/OpenLiberty/guides-common/issues/540

Added a new section to explain and demonstrate externalizing server configuration.

AustinSeto commented on README.adoc:
````diff
@@ -312,7 +383,7 @@ Run the Maven `package` goal to compile the test classes. Run the Maven `failsaf
 ```
 mvn package
-mvn failsafe:integration-test -Dsystem.ip=[system-ip-address] -Dinventory.http.port=9081 -Dsystem.http.port=9080
+mvn failsafe:integration-test -Dsystem.ip=[system-ip-address] -Dinventory.http.port=9091 -Dsystem.http.port=9080
````
I get the following error when running this command on Windows

```
[ERROR] Unknown lifecycle phase ".ip=172.17.0.2". You must specify a valid lifecycle phase or a goal in the format <plugin-prefix>:<goal> or <plugin-group-id>:<plugin-artifact-id>[:<plugin-version>]:<goal>. 
```

griffinhadfield replied:
I discussed with Gilbert and he said to leave the command as is. The guides don't claim to support PowerShell so we don't need to work around it.

AustinSeto reviewed (changes requested):
In the section "Running the tests" it says to use the IP address determined in the previous section. However, with the addition of the externalizing section, this sentence no longer applies, as the `docker inspect -f "{{.NetworkSettings.IPAddress }}" system` command was run two sections prior. I think it might be worth it to ask the user to run this command again in this section.

griffinhadfield commented:
> In the section "Running the tests" it says to use the IP address determined in the previous section. However, with the addition of the externalizing section, this sentence no longer applies, as the `docker inspect -f "{{.NetworkSettings.IPAddress }}" system` command was run two sections prior. I think it might be worth it to ask the user to run this command again in this section.

@AustinSeto I adjusted the wording so that it is still referenced accurately. I didn't add the command in again because there doesn't seem to be a way to do it with making it awkward. I think it should be clear enough to the user that they should scroll up or re-run the command.

AustinSeto reviewed (approved):
LGTM

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+include::finish/inventory/src/main/liberty/config/server.xml[]
+----
+
+As mentioned in the beginning of this guide, one of the advantages of using
```
Change,
"As mentioned in the beginning of this guide"
to,
"As mentioned at the beginning of this guide"

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+As mentioned in the beginning of this guide, one of the advantages of using
+containers is that they are portable and can be moved and deployed efficiently
+across all of your DevOps environments. Configuration often changes across
+different environments, and by externalizing your server's configuration, you
```
Change,
"and by externalizing your server's configuration"
to,
"and by externalizing your server configuration"

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+different environments, and by externalizing your server's configuration, you
+can simplify the development process.
+
+Imagine a scenario where you are developing an Open Liberty application on
```
Change,
"Imagine a scenario where you are developing an Open Liberty application on
port `9081` but when it comes time to deploy it to production, it must be available
on port `9091`."
to,
"Imagine a scenario where you are developing an Open Liberty application on
port `9081` but to deploy it to production, it must be available
on port `9091`."

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+
+Imagine a scenario where you are developing an Open Liberty application on
+port `9081` but when it comes time to deploy it to production, it must be available
+on port `9091`. To manage this, you could keep two different versions of the
```
Change,
"To manage this, you could keep two different versions of the
`server.xml` file; one for production and one for development."
to,
"To manage this scenario, you can keep two different versions of the
`server.xml` file; one for production and one for development."

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+Imagine a scenario where you are developing an Open Liberty application on
+port `9081` but when it comes time to deploy it to production, it must be available
+on port `9091`. To manage this, you could keep two different versions of the
+`server.xml` file; one for production and one for development. However, trying to
```
Change,
" However, trying to maintain two different versions of a file may lead to mistakes."
to,
" However, trying to
maintain two different versions of a file might lead to mistakes."

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+In this example, you will use an environment variable to externally configure the
+HTTP port number of the `inventory` service. 
+
+In the [hotspot file=0]`inventory/server.xml` file, the [hotspot=httpPort
```
Change,
"In the [hotspot file=0]`inventory/server.xml` file, the [hotspot=httpPort
file=0]`default.http.port` variable is declared and is used in the
[hotspot=httpEndpoint file=0]`httpEndpoint` element to define the service's
endpoint."
to,
"In the [hotspot file=0]`inventory/server.xml` file, the [hotspot=httpPort
file=0]`default.http.port` variable is declared and is used in the
[hotspot=httpEndpoint file=0]`httpEndpoint` element to define the service
endpoint. "

ManasiGandhi replied:
Change,
"In the [hotspot file=0]`inventory/server.xml` file, the [hotspot=httpPort
file=0]`default.http.port` variable is declared and is used in the
[hotspot=httpEndpoint file=0]`httpEndpoint` element to define the service's
endpoint."
to,
"In the [hotspot file=0]`inventory/server.xml` file, the [hotspot=httpPort
file=0]`default.http.port` variable is declared and is used in the
[hotspot=httpEndpoint file=0]`httpEndpoint` element to define the service
endpoint."

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+In the [hotspot file=0]`inventory/server.xml` file, the [hotspot=httpPort
+file=0]`default.http.port` variable is declared and is used in the
+[hotspot=httpEndpoint file=0]`httpEndpoint` element to define the service's
+endpoint. The default value of the [hotspot=httpPort file=0]`default.http.port`
```
Change,
" The default value of the [hotspot=httpPort file=0]`default.http.port`
variable is `9081`, however this value is only used if no other value is
specified."
to,
" The default value of the [hotspot=httpPort file=0]`default.http.port`
variable is `9081`. However, this value is only used if no other value is
specified."

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+[hotspot=httpEndpoint file=0]`httpEndpoint` element to define the service's
+endpoint. The default value of the [hotspot=httpPort file=0]`default.http.port`
+variable is `9081`, however this value is only used if no other value is
+specified. To find a value for this variable, Open Liberty will look for the
```
Change,
"To find a value for this variable, Open Liberty will look for the
following environment variables, in order:"
to,
"To find a value for this variable, Open Liberty looks for the
following environment variables, in order:"

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+* `default_http_port`
+* `DEFAULT_HTTP_PORT`
+
+When you previously ran the `inventory` container, none of the above environment variables were defined and thus the default value of `9081` was used.
```
Change,
"When you previously ran the `inventory` container, none of the above environment variables were defined and thus the default value of `9081` was used."
to,
"When you previously ran the `inventory` container, none of the environment variables mentioned were defined and thus the default value of `9081` was used."

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+in a Docker container. In this case, you are setting the `default.http.port` environment
+variable to `9091` for the `inventory` container.
+
+Now, when the service is starting up, Open Liberty will find the
```
Change,
"Now, when the service is starting up, Open Liberty will find the
`default.http.port` environment variable and use it to set the value of the
[hotspot=httpPort file=0]`default.http.port` variable to be used in the HTTP
endpoint."
to,
"Now, when the service is starting up, Open Liberty finds the
`default.http.port` environment variable and use it to set the value of the
[hotspot=httpPort file=0]`default.http.port` variable to be used in the HTTP
endpoint."

griffinhadfield replied:
Could you double check what you meant for me to say? Did you want me to say finds/uses or will find/use?

Thanks!

ManasiGandhi replied:
@griffinhadfield Sorry for that. Updated the comment.

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+endpoint.
+
+The `inventory` service will now be available on the new port number you
+specified. You should be able to see the contents of the inventory at
```
Change,
"You should be able to see the contents of the inventory at
http://localhost:9091/inventory/systems[http://localhost:9091/inventory/systems^]."
to,
"You can see the contents of the inventory at
http://localhost:9091/inventory/systems[http://localhost:9091/inventory/systems^].

ManasiGandhi reviewed (changes requested):
Few edit suggestions.

ManasiGandhi reviewed (changes requested):
Few edit suggestions.

ManasiGandhi commented on README.adoc:
```diff
@@ -269,6 +274,78 @@ properties are automatically stored in the inventory. Go back to http://localhos
+http://localhost:9080/system/properties[http://localhost:9080/system/properties^]
+
+You can externalize the configuration of more than just the port numbers.
+To learn more about Open Liberty server configuration, check out the
```
Change,
"To learn more about Open Liberty server configuration, check out the
https://openliberty.io/docs/20.0.0.11/reference/config/server-configuration-overview.html[Server
Configuration Overview] docs."
to,
"To learn more about Open Liberty server configuration, check out the
https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html[Server
Configuration Overview] docs. "

ManasiGandhi reviewed (changes requested):
Please update the link

griffinhadfield commented:
@ManasiGandhi Thanks for the feedback, I have addressed your comment and pushed the changes. The updated content can be seen on lgdev here: http://lgdev1.fyre.ibm.com:4020/guides/containerize

gkwan-ibm reviewed (changes requested):
pending on all reviews
